### PR TITLE
Sync IDisk.h

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit c518d8acb1b30fa71b809d8ecf83b99d5200ebce with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64839/c518d8acb1b30fa71b809d8ecf83b99d5200ebce/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Disks/IDisk.h [hunk 1]
```diff
@@ -564,6 +549,27 @@ inline String directoryPath(const String & path)
...
+    path = normalizePath(path);
+
+    return path.starts_with(prefix) &&
+        (path.size() == prefix.size() || path[prefix.size()] == '/');
+}
+
+inline bool arePathsEqual(std::string_view lhs, std::string_view rhs)
+{
+    return normalizePath(lhs) == normalizePath(rhs);
+}
```
Comment (vdimir) [inline src/Disks/IDisk.h:571]: Probably this will not be needed

```suggestion
```

**Comment by vdimir** [comment]
```
Jun 05 14:03:48 In file included from /build/src/Disks/ObjectStorages/DiskObjectStorageCache.cpp:3:
Jun 05 14:03:48 /build/src/Disks/ObjectStorages/DiskObjectStorage.h:216:66: error: only virtual member functions can be marked 'override'
Jun 05 14:03:48   216 |     std::shared_ptr<const S3::Client> getS3StorageClient() const override;
Jun 05 14:03:48       |                                                                  ^~~~~~~~
Jun 05 14:03:48 1 error generated.
```

File: src/Disks/IDisk.h (hunk 1 above)
Comment (vdimir) [reply src/Disks/IDisk.h:586]: Actually it's the only difference, since other changes from outdated version, so let's not to fix it here

**Pull request closed without merging**